src/Microsoft.AspNetCore.WebHooks.Receivers/Filters/WebHookVerifyBodyTypeFilter.cs

```diff
@@ -125,6 +132,70 @@ public void OnResourceExecuted(ResourceExecutedContext context)
             // No-op
         }
 
+        /// <summary>
+        /// Determines whether the specified request contains JSON as indicated by a content type of
+        /// <c>application/json</c>, <c>text/json</c> or <c>application/xyz+json</c>. The term <c>xyz</c> can for
+        /// example be <c>hal</c> or some other JSON-derived media type.
+        /// </summary>
+        /// <param name="request">The <see cref="HttpRequest"/> to check.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified request contains JSON content; otherwise, <see langword="false"/>.
+        /// </returns>
+        protected static bool IsJson(HttpRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var contentType = request.GetTypedHeaders()?.ContentType;
+            if (contentType == null)
+            {
+                return false;
+            }
+
+            if (contentType.IsSubsetOf(ApplicationJsonMediaType) || contentType.IsSubsetOf(TextJsonMediaType))
+            {
+                return true;
+            }
+
+            // MVC's JsonInputFormatter does not support text/*+json by default.
+            return contentType.Type.Equals("application", StringComparison.OrdinalIgnoreCase) &&
+                contentType.SubType.EndsWith("+json", StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Determines whether the specified request contains XML as indicated by a content type of
+        /// <c>application/xml</c>, <c>text/xml</c> or <c>application/xyz+xml</c>. The term <c>xyz</c> can for example
+        /// be <c>rdf</c> or some other XML-derived media type.
+        /// </summary>
+        /// <param name="request">The <see cref="HttpRequest"/> to check.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified request contains XML content; otherwise, <see langword="false"/>.
+        /// </returns>
+        protected static bool IsXml(HttpRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var contentType = request.GetTypedHeaders()?.ContentType;
+            if (contentType == null)
+            {
+                return false;
+            }
+
+            if (contentType.IsSubsetOf(ApplicationXmlMediaType) || contentType.IsSubsetOf(TextXmlMediaType))
+            {
+                return true;
+            }
+
+            // MVC's XML input formatters do not support text/*+xml by default.
```

Comment:
FYI, this is because the [RFC](https://tools.ietf.org/html/rfc6839#section-3.1) doesn't mention text/json and text/xml. The suffix content type feature specifically applies to application/json and application/xml (among others) specifically

Reply:
FYI, I know 😈 I added this comment because the ASP.NET WebHooks code checks for the (non-existent) `text/*+json` and `text/*+xml` cases. Don't want some later Doug to make them consistent. Of, if we do make them consistent, change the older code.